### PR TITLE
Delay workflow email cancellation in sendgrid for all case

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Sendgrid's limit for entries in` /scheduled_sends `for cancelled scheduled emails got reached again. After making some first improvements in #7232, we need to make sure to keep the number of cancelled `scheduled_sends` lower (as low as possible).

With this PR `WorkflowStep` and `Booking` will now be optional fields in `WorkflowReminder`, so we can get rid of the immediate delete that was done when workflow or workflow step was deleted and also when an event was rescheduled.

### CODE REVIEW

File: packages/features/ee/workflows/api/scheduleSMSReminders.ts [hunk 1]
```diff
@@ -53,83 +54,85 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
   if (!unscheduledReminders.length) res.json({ ok: true });
 
   for (const reminder of unscheduledReminders) {
-    try {
-      const sendTo =
-        reminder.workflowStep.action === WorkflowActions.SMS_NUMBER
-          ? reminder.workflowStep.sendTo
-          : reminder.booking?.smsReminderNumber;
+    if (reminder.workflowStep && reminder.booking) {
```
Comment (emrysal) [inline packages/features/ee/workflows/api/scheduleSMSReminders.ts:57]: Invert and continue; - you're going to deep 🙏

Comment (CarinaWolli) [reply packages/features/ee/workflows/api/scheduleSMSReminders.ts]: Fixed, and in scheduleEmailReminder as well

File: packages/prisma/migrations/20230309203435_make_booking_and_workflow_step_optional_for_workflow_reminder/migration.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+-- DropForeignKey
+ALTER TABLE "WorkflowReminder" DROP CONSTRAINT "WorkflowReminder_bookingUid_fkey";
+
+-- DropForeignKey
+ALTER TABLE "WorkflowReminder" DROP CONSTRAINT "WorkflowReminder_workflowStepId_fkey";
+
+-- AlterTable
+ALTER TABLE "WorkflowReminder" ALTER COLUMN "bookingUid" DROP NOT NULL,
+ALTER COLUMN "workflowStepId" DROP NOT NULL;
```
Comment (emrysal) [inline packages/prisma/migrations/20230309203435_make_booking_and_workflow_step_optional_for_workflow_reminder/migration.sql:9]: Checked there are no NULL variables on production for `bookingUid` and `workflowStepId`

**Review by emrysal — APPROVED** [review]
LGTM 👍

**Pull request merged**